packages/validation/src/validators/LionWebLanguageValidator.ts

```diff
@@ -1,39 +1,84 @@
-            this.validationResult.issue(new IncorrectLionCoreVersion_Issue(new JsonContext(null, ["languages", 0]), usedLanguage.version))
+    validateLanguage(chunk: LionWebJsonChunk) {
+        this.chunkWrapper = new LionWebJsonChunkWrapper(chunk)
+        const usedM3Language = chunk.languages.find(lang => isLionWebM3Language(lang))
+        if (usedM3Language === undefined) {
+            this.validationResult.issue(new MissingM3Language_Issue(new JsonContext(null, ["languages"])))
         }
-        const chunkWrapper = new LionWebJsonChunkWrapper(chunk)
-        const languageNodes = chunkWrapper.findNodesOfConcept("Language")
+
+        const languageNodes = this.chunkWrapper.findNodesOfClassifier(MetaPointers.Language)
         if (languageNodes.length !== 1) {
             // TODO Better error handling.
-            console.error("Expected exactly one Language node, found " + languageNodes.length + " => " + JSON.stringify(languageNodes))
+            console.error("Error: xpected exactly one Language node, found " + languageNodes.length + " => " + JSON.stringify(languageNodes))
         }
+        chunk.nodes.forEach((node, index) => {
+            if (!isConcept(node)) {
+                this.validationResult.issue(new GenericIssue(new JsonContext(null, ["nodes", index]), `node ${node.id} is not a concept`))
+            } else {
+                this.validateConcept(node)
+            }
+        })
+    }
+
+    validateConcept(node: LionWebJsonNode): void {
+        node.properties.forEach(prop => {
+            const properties = this.chunkWrapper?.findNodesOfClassifier(MetaPointers.Property)
+            const matchedProperty = properties?.find(p => isEqualMetaPointer(p.classifier, prop.property))
+            // DUMMY
+            return matchedProperty !== undefined
+        })
+    }
+
+    validateNode(node: LionWebJsonNode): void {
+        const classifier = this.registry.getNodeByMetaPointer(node.classifier)
+        if (classifier === undefined) {
+            this.validationResult.issue(
+                new GenericIssue(new JsonContext(null, ["nodes"]), `Classifier ${node.classifier.key} not found for node ${node.id}`)
+            )
+            return
+        }
+        this.validateProperties(node, classifier)
+    }
+    
+    validateProperties(node: LionWebJsonNode, classifier: LionWebJsonNode): void {
+        node.properties.forEach(actualProp => {
+            const propClassifier = this.registry.getNodeByMetaPointer(actualProp.property)
+            if (propClassifier === undefined) {
+                this.validationResult.issue(
+                    new GenericIssue(new JsonContext(null, ["nodes", "properties"]), 
+                        `Property ${actualProp.property.key} not found for classifier ${classifier.id}`)
+                )
+            }
+            // this.availableLanguages.getClassifierProperty(classifier, propClassifier)
+        })
+        // node.containments
```

Comment:
Do these comments mean they still need to be implemented?

Reply:
No, will remove

Reply:
Actually the validation is not complete yet, but these comment are out of date.